build-aux/m4/gmp.m4

```diff
@@ -0,0 +1,16 @@
+dnl Copyright (c) 2018 The PIVX Core developers
+dnl Distributed under the MIT software license, see the accompanying
+dnl file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+dnl
+AC_DEFUN([GMP_CHECK],[
+if test x"$has_gmp" != x"yes"; then
+  CPPFLAGS_TEMP="$CPPFLAGS"
+  CPPFLAGS="$GMP_CPPFLAGS $CPPFLAGS"
+  LIBS_TEMP="$LIBS"
+  LIBS="$GMP_LIBS $LIBS"
```

Comment:
I know this has passed travis tests as-is, but the ordering of things here seem quite off. Setting `LIBS="$GMP_LIBS $LIBS"` **before** the header and lib check, and then re-setting `LIBS` further down (without including `$GMP_LIBS`) just reverts the variable no?